src/olympia/accounts/tests/test_utils.py

```diff
@@ -111,3 +116,70 @@ def test_redirect_for_login(default_fxa_login_url):
     response = utils.redirect_for_login(request)
     default_fxa_login_url.assert_called_with(request)
     assert response['location'] == login_url
+
+
+class TestProcessSqsQueue(TestCase):
+
+    @mock.patch('olympia.accounts.utils.process_fxa_event')
+    @mock.patch('boto3.client')
+    def test_process_sqs_queue(self, client, process_fxa_event):
+        messages = [{'Body': 'foo', 'ReceiptHandle': '$$$'}, {'Body': 'bar'}]
+        sqs = mock.MagicMock(
+            **{'receive_message.side_effect': [{'Messages': messages}]})
+        delete_mock = mock.MagicMock()
+        sqs.delete_message = delete_mock
+        client.return_value = sqs
+
+        with self.assertRaises(StopIteration):
+            utils.process_sqs_queue(
+                queue_url='https://nowh.ere/', aws_region='NARNIA_1',
+                queue_wait_time=20)
+
+        client.assert_called()
+        process_fxa_event.assert_called()
+        process_fxa_event.assert_has_calls(
+            [mock.call('foo'), mock.call('bar')])
+        delete_mock.assert_called_once()  # Receipt handle is present in foo.
+        delete_mock.assert_called_with(
+            QueueUrl='https://nowh.ere/', ReceiptHandle='$$$')
```

Comment:
You could try using `moto` - https://github.com/spulec/moto instead of mocking all the things and get more realistic tests.

Looks good otherwise though

Reply:
I tried; I failed 😿